### PR TITLE
Permit binding of joystick axes to look up/down

### PR DESCRIPTION
**Pull request by jmtd**
In games which support a look/down action (Heretic, Hexen, Strife)
permit binding a joystick axis to those actions.

Altazimuth was asking me some questions on IRC about how we handle
mouse wheel actions and that got me thinking about some of the
input stuff (specifically joystick stuff) that we hadn't finished
off.

### CODE REVIEW

File: src/setup/joystick.c [hunk 1]
```diff
@@ -1021,9 +1028,25 @@ void ConfigJoystick(void)
...
+                                       &joystick_look_invert,
+                                        JOYSTICK_AXIS_VERTICAL),
+                   TXT_TABLE_OVERFLOW_RIGHT,
+                   TXT_TABLE_OVERFLOW_RIGHT,
+                   TXT_TABLE_EMPTY,
+                   TXT_TABLE_EMPTY,
                    NULL);
+    }
+
+    TXT_AddWidgets(window,
```
Comment (jmtd) [inline src/setup/joystick.c]: I guess this could just be `TXT_AddWidget` now

File: src/setup/joystick.c [hunk 2]
```diff
@@ -1021,9 +1028,25 @@ void ConfigJoystick(void)
                    TXT_TABLE_OVERFLOW_RIGHT,
                    TXT_TABLE_EMPTY,
                    TXT_TABLE_EMPTY,
+                   NULL);
 
-                   TXT_NewSeparator("Buttons"),
+    if (gamemission >= heretic) // heretic, hexen, strife
```
Comment (jmtd) [inline src/setup/joystick.c]: expecting some resistance to this (but I like it)

File: src/d_event.h [hunk 3]
```diff
@@ -72,7 +72,7 @@ typedef struct
     evtype_t type;
 
     // Event-specific data; see the descriptions given above.
```
Comment (jmtd) [inline src/d_event.h]: oh, I haven't added a description of the meaning of data5 in the comment above.

**Comment by jmtd** [comment]
It feels *really* nice to have this action bound to a perpendicular axis to turn left/right when using a joystick/gamepad.

File: src/setup/joystick.c (hunk 2 above)
Comment (chungy) [reply src/setup/joystick.c]: What if it is instead `if (gamemission != doom)`?

Comment (fragglet) [reply src/setup/joystick.c]: I prefer `gamemission == heretic || gamemission == hexen || gamemission == strife`.

A good way to think about it is to ask: if another game was added, what would happen?

Comment (jmtd) [reply src/setup/joystick.c]: I've gone with @fragglet suggestion. I'm just a sucker for trying to squeeze more utility out of enums...

**Pull request merged**